### DISCUSSION TITLE
How do i tween a Color?

### DISCUSSION THREAD

**Discussion by ZedDevStuff** [KyryloKuzyk/PrimeTween#53] [FAQ] [answered]
Tween.Color only seems to be able to target UI elements and others with a color field but i just need to tween a color directly (for UI Toolkit stuff)

**Comment by KyryloKuzyk** [comment, marked as answer]
There is currently no built-in support for animating UIToolkit color, but it's possible to animate anything with the Custom animation:
```csharp
Color startValue = visualElement.style.color.value;
Color endValue = Color.red;
Tween.Custom(visualElement, startValue, endValue, duration, (target, val) => target.style.color = val);
```

Currently, only [these](https://docs.unity3d.com/ScriptReference/UIElements.Experimental.ITransitionAnimations.html) properties are supported. I'll add support for UIToolkit's color and backgroundColor in the next PrimeTween version. Please tell me what other UIToolkit properties you would like to see.

**Comment by KyryloKuzyk** [comment]
I added Tween.VisualElementColor/VisualElementBackgroundColor() methods in version **1.1.14** to animate colors of UIToolkit components.
Please update to the latest version :)
